microprofile/telemetry/src/main/java/io/helidon/microprofile/telemetry/OtelProxy.java

```diff
@@ -0,0 +1,263 @@
+        allInterfaces[allInterfaces.length - 1] = ProxiedTracer.class;
+        return (Tracer) Proxy.newProxyInstance(OtelProxy.class.getClassLoader(),
+                                               allInterfaces,
+                                               new TracerHandler(otelTracer, HelidonOpenTelemetry.create(openTelemetry,
+                                                                                                         otelTracer,
+                                                                                                         Map.of())));
+    }
+
+    static Span span(io.helidon.tracing.Span.Extended<Scope> helidonSpan) {
+        Span delegate = helidonSpan.unwrap(Span.class);
+        return (Span) Proxy.newProxyInstance(OtelProxy.class.getClassLoader(),
+                                             delegate.getClass().getInterfaces(),
+                                             new SpanHandler(delegate, helidonSpan));
+    }
+
+    interface ProxiedTracer {
+        io.helidon.tracing.Tracer helidonTracer();
+    }
+
+    private static SpanBuilder spanBuilder(io.helidon.tracing.Span.Builder<?> helidonSpanBuilder) {
+        SpanBuilder delegate = helidonSpanBuilder.unwrap(SpanBuilder.class);
+        return (SpanBuilder) Proxy.newProxyInstance(OtelProxy.class.getClassLoader(),
+                                                    delegate.getClass().getInterfaces(),
+                                                    new SpanBuilderHandler(delegate, helidonSpanBuilder));
+    }
+
+    private static Scope scope(io.helidon.tracing.Scope helidonScope,
+                               Scope delegate) {
+        return (Scope) Proxy.newProxyInstance(OtelProxy.class.getClassLoader(),
+                                              delegate.getClass().getInterfaces(),
+                                              new ScopeHandler(delegate, helidonScope));
+    }
+
+    private OtelProxy() {
+    }
+
+    private record OpenTelemetryHandler(OpenTelemetry openTelemetry) implements InvocationHandler {
+
+            private static final Method TRACER_WITH_NAME;
+            private static final Method TRACER_WITH_NAME_AND_VERSION;
+
+            static {
+                try (MethodHelper methodHelper = new MethodHelper(OpenTelemetry.class)) {
+                    TRACER_WITH_NAME = methodHelper.method("getTracer", String.class);
+                    TRACER_WITH_NAME_AND_VERSION = methodHelper.method("getTracer", String.class, String.class);
+                }
+            }
+
+        @Override
+            public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+                if (TRACER_WITH_NAME.equals(method) || TRACER_WITH_NAME_AND_VERSION.equals(method)) {
+                    return OtelProxy.tracer(openTelemetry, (Tracer) method.invoke(openTelemetry, args));
+                } else {
+                    return method.invoke(openTelemetry, args);
```

Comment:
You'll also want to [treat `equals` and `hashCode` methods very specially](https://javaclippings.wordpress.com/2009/03/18/dynamic-proxies-equals-hashcode-tostring/).